Products/CMFPlone/recyclebin.py

```diff
@@ -166,6 +169,104 @@ def is_enabled(self):
         except (KeyError, AttributeError):
             return False
 
+    def check_permission(self, check_roles=True):
+        """Check if the current user has permission to manage the recycle bin
+
+        Args:
+            check_roles: If True, also check for Manager or Site Administrator roles
+
+        Returns:
+            Boolean indicating permission
+        """
+        context = self._get_context()
+        sm = getSecurityManager()
+
+        # Check for explicit permission first
+        has_permission = sm.checkPermission(self.MANAGE_RECYCLEBIN, context)
+        if has_permission:
+            return True
+
+        # If no explicit permission and we're not checking roles, return False
+        if not check_roles:
+            return False
+
+        # Additional check for Manager or Site Administrator roles
+        user = sm.getUser()
+        user_roles = user.getRolesInContext(context)
+        return "Manager" in user_roles or "Site Administrator" in user_roles
+
+    def can_restore(self, item_id):
+        """Check if the current user can restore an item
+
+        A user can restore an item if:
+        1. They have the ManageRecycleBin permission, or
+        2. They are a Manager or Site Administrator, or
+        3. They are the owner of the item, or
+        4. They deleted the item
```

Comment:
3 and 4 are a nice idea, but currently the "Manage recycle bin" permission is only granted to Managers and Site Administrators, which means that the recycle bin can't be accessed by users who don't have one of those roles.

There are a lot of details to get right here. I strongly suggest we focus on making this work only for Managers and Site Administrators for now, and then come back to it later if we want to provide more filtered access to other users.

Reply:
done
will soon create an issue for tracking